fuel-txpool/src/txpool.rs

```diff
@@ -132,6 +132,128 @@ impl TxPool {
+                        // small todo there is possibility to have removal reason (ReplacedByHigherGas, DependencyRemoved)
+                        // but for now it is okay to just use Error::Removed.
+                        let _ = broadcast.send(TxStatusBroadcast {
+                            tx: removed.clone(),
+                            status: TxStatus::SqueezedOut {
+                                reason: Error::Removed,
+                            },
+                        });
+                    }
+                    let _ = broadcast.send(TxStatusBroadcast {
+                        tx,
+                        status: TxStatus::Submitted,
+                    });
+                }
+                Err(_) => {}
+            }
+        }
+        res
+    }
+
+    /// find all tx by its hash
+    pub async fn find(txpool: &RwLock<Self>, hashes: &[TxId]) -> Vec<Option<TxInfo>> {
+        let mut res = Vec::with_capacity(hashes.len());
+        let pool = txpool.read().await;
+        for hash in hashes {
+            res.push(pool.txs().get(hash).cloned());
+        }
+        res
+    }
+
+    pub async fn find_one(txpool: &RwLock<Self>, hash: &TxId) -> Option<TxInfo> {
+        txpool.read().await.txs().get(hash).cloned()
+    }
+
+    /// find all dependent tx and return them with requsted dependencies in one list sorted by Price.
+    pub async fn find_dependent(txpool: &RwLock<Self>, hashes: &[TxId]) -> Vec<ArcTx> {
+        let mut seen = HashMap::new();
+        {
+            let pool = txpool.read().await;
+            for hash in hashes {
+                if let Some(tx) = pool.txs().get(hash) {
+                    pool.dependency()
+                        .find_dependent(tx.tx().clone(), &mut seen, pool.txs());
+                }
+            }
+        }
+        let mut list: Vec<ArcTx> = seen.into_iter().map(|(_, tx)| tx).collect();
+        // sort from high to low price
+        list.sort_by_key(|tx| Reverse(tx.gas_price()));
+
+        list
+    }
+
+    /// Iterete over `hashes` and return all hashes that we dont have.
+    pub async fn filter_by_negative(txpool: &RwLock<Self>, tx_ids: &[TxId]) -> Vec<TxId> {
+        let mut res = Vec::new();
+        let pool = txpool.read().await;
+        for tx_id in tx_ids {
+            if pool.txs().get(tx_id).is_none() {
+                res.push(*tx_id)
+            }
+        }
+        res
+    }
+
+    /// Return all sorted transactions that are includable in next block.
+    /// This is going to be heavy operation, use it only when needed.
+    pub async fn includable(txpool: &RwLock<Self>) -> Vec<ArcTx> {
+        let pool = txpool.read().await;
+        pool.sorted_includable()
+    }
+
+    /// When block is updated we need to receive all spend outputs and remove them from txpool.
+    pub async fn block_update(
+        txpool: &RwLock<Self>, /*spend_outputs: [Input], added_outputs: [AddedOutputs]*/
+    ) {
+        txpool.write().await;
+        // TODO
```

Comment:
What is this TODO still for?

Reply:
It is pending task, created github issue for it https://github.com/FuelLabs/fuel-core/issues/465. I didn't know at that time how exactly this will look.